app/webpacker/scripts/live_search.js

```diff
@@ -0,0 +1,61 @@
+import $ from 'jquery'
+
+export default class LiveSearch {
+  static init () {
+    return new LiveSearch()
+  }
+
+  constructor () {
+    this.form = document.querySelector('#js-live-filter')
+    this.resultsDiv = document.querySelector('#js-results')
+    this.selectedFiltersDiv = document.querySelector('#js-selected-filters')
+    this.traineeCount = document.querySelector('#js-trainee-count')
+
+    this.state = null
+
+    if (!(this.form && this.resultsDiv && this.selectedFiltersDiv)) return
+
+    this.hideButton()
+    this.bindEvents()
+  }
+
+  updateUrl () {
+    const newPath = window.location.pathname + '?' + $.param(this.state)
+    window.history.pushState(this.state, '', newPath)
+  }
+
+  displayResults (response) {
+    this.resultsDiv.innerHTML = response.results
+    this.selectedFiltersDiv.innerHTML = response.selected_filters
+    this.traineeCount.innerHTML = response.trainee_count
+  }
+
+  updatePage () {
+    return $.ajax({
+      url: '/trainees',
+      dataType: 'json',
+      data: this.state
+    }).done(response => { this.displayResults(response) })
```

Comment:
```suggestion
    }).done(this.displayResults)
```

Reply:
i do this every. single. time. will i ever learn?

Reply:
needs a `bind` though 😞